subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftApplicationPlugin.java

```diff
@@ -70,5 +78,27 @@ public void apply(final ProjectInternal project) {
 
         // Wire in this install task
         tasks.getByName(LifecycleBasePlugin.ASSEMBLE_TASK_NAME).dependsOn(application.getDevelopmentBinary().getInstallDirectory());
+
+        // Configure compile task
+        final SwiftCompile compileDebug = (SwiftCompile) tasks.getByName("compileDebugSwift");
+        final SwiftCompile compileRelease = (SwiftCompile) tasks.getByName("compileReleaseSwift");
+
+        Configuration implementation = application.getImplementationDependencies();
+
+        Configuration debugApiElements = configurations.maybeCreate("debugSwiftApiElements");
```

Comment:
An application doesn't provide a Swift API, otherwise it's a library. We shouldn't be adding these configurations.

Maybe just merge this for now and we can remove these later.

Reply:
I thought the same thing when I was writing this and would have wanted to discuss it with you. The issue I was having is when we add a dependency from the test component on the main component, the `artifactView` wasn't filtering out the main component because the way we used to add it didn't add a `ComponentIdentifier`. This ended up compiling the code from the main component in if XCTest plugin was applied. My thinking is this allow someone to add a test on the main component in a different project. I'm not sure if this is something we want, but it's something that is valid. I will open an issue for this.

Reply:
Issue created: https://github.com/gradle/gradle-native/issues/333.